libcodechecker/server/api/report_server.py

```diff
@@ -1715,6 +1714,20 @@ def removeRunResults(self, run_ids):
                     LOG.debug("Can't delete " + str(run_id))
                     continue
 
+                # Check if there is an existing lock on the given run name,
+                # which has not expired yet. If so, the run cannot be
+                # deleted, as someone is assumed to be storing into it.
+                locks_expired_at = datetime.now() - \
+                    timedelta(seconds=db_cleanup.RUN_LOCK_TIMEOUT_IN_DATABASE)
+                lock = session.query(RunLock) \
+                    .filter(RunLock.name == run_to_delete.name,
+                            RunLock.locked_at >= locks_expired_at) \
+                    .one_or_none()
+                if lock:
+                    LOG.info("Can't delete '{0}' as it is locked."
+                             .format(run_to_delete.name))
+                    continue
```

Comment:
I stored a run which failed so the lock is still exists on the run. If a try to remove the run in the GUI, the run is disappears but if I refresh the page the run is still exists. 

It would be great to give an error message on the GUI that some of the runs couldn't be removed and do not remove these runs from the Run list table too.

Reply:
I think this issue should be fixed separately, as this concerns the `can_delete` variable of a Run record, and the whole logic behind that particular variable. This extension here should be good enough to keep the functionality logical.

Reply:
Let's create a separate issue for this problem then.